Sources/TokamakLint/TokenVisitor.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  TokenVisitor.swift
+//  SwiftSyntax
+//
+//  Created by Matvii Hodovaniuk on 4/2/19.
+//
+
+import Foundation
+import SwiftSyntax
+
+class TokenVisitor: SyntaxVisitor {
+  public var tree = [Node]()
+  public var current: Node!
+
+  var row = 0
+  var column = 0
+
+  override func visitPre(_ node: Syntax) {
+    var syntax = "\(type(of: node))"
+    if syntax.hasSuffix("Syntax") {
+      syntax = String(syntax.dropLast(6))
+    }
+
+    let node = Node(text: syntax)
+    node.range.startRow = row
+    node.range.startColumn = column
+    node.range.endRow = row
+    node.range.endColumn = column
+    if current == nil {
+      tree.append(node)
+    } else {
+      current.add(node: node)
+    }
+    current = node
+  }
+
+  override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
+    current.text = escapeHtmlSpecialCharacters(token.text)
+    current.token = Node.Token(
+      kind: "\(token.tokenKind)",
+      leadingTrivia: "",
+      trailingTrivia: ""
+    )
+
+    current.range.startRow = row
+    current.range.startColumn = column
+
+    token.leadingTrivia.forEach { piece in
+      let trivia = processTriviaPiece(piece)
+      current.token?.leadingTrivia += replaceSymbols(text: trivia)
+    }
+    processToken(token)
+    token.trailingTrivia.forEach { piece in
+      let trivia = processTriviaPiece(piece)
+      current.token?.trailingTrivia += replaceSymbols(text: trivia)
+    }
```

Comment:
I'm not sure why this trivia has to be processed for the checks, could you please add code comments that clarify the purpose of this in details?